### ISSUE TITLE
Cannot get moonlight to work with nvidia shield pro

### ISSUE THREAD

**Issue by jasonc-87** [moonlight-stream/moonlight-android#925]
Cannot get moonlight or gamestream working with NVIDIA Shield Pro.

I have followed the setup directions for moonlight.

Both NVIDIA shield pro and PC are hardwired.

Geforce experience has gamestream enabled obviously.

I've opened all the ports listed in the throubleshooting.

I do have bitdefender which I've completely uninstalled and disabled windows firewall settings.

My PC is not detected on the moonlight app, I try to enter the IP manually and it says I can't connect + check my ports. If everything is uninstalled and disabled I don't think it would be a port problem at that point.

I do have a google fiber router and a google mesh system(not sure that either of those things matter)

I've factory reset the nvidia shield pro, reinstalled geforce experience. 

Still the same error and cannot connect with moonlight. If I try gamestream it gives me a 80020003 which is port/firewall related and I've tried all of those steps plus the above ones because I figure it's related to moonlight issue.

Any thoughts besides just give up?

**Comment by ReenigneArcher** [comment]
Is the PC locked? (Moonlight won't connect to a locked PC, although I think GeForce now will)
Do you have a monitor connected to your PC? (Game stream won't work on headless PCs)
Are you connected to your PC using a RDP session? (Game stream won't work while Remote Desktop is active)

**Comment by jasonc-87** [comment]
Figured out the issue and resolved it.  It had to do with a double NAT caused by a google mesh system/MoCA box setup since there is no ethernet throughout the house.  Here is the fix if anyone is interested, there is not much documentation on it.

The router from google fiber is NAT(cannot be changed to bridged connection, lets call this NAT1).  The main mesh puck is also NAT(couldn't bridge it either, lets call it NAT2)  and basically acts as a second router, causing a double NAT issue.  I hard wired the main puck into the router and did not feed anything off of it.  I ran an ethernet from router to nvidia shield pro(NAT1).  I ran another ethernet to my MoCA box/coax connection which feeds to another room MoCA box and mesh puck(this isn't the primary and it will use whatever NAT feeds it, so it's still on NAT1).  This resolved my issue, I still have 2 "hard wired" mesh pucks but my PC and shield are on the same NAT so my issue has been resolved.
